modules/utils/context.go

```diff
@@ -0,0 +1,50 @@
+package utils
```

Comment:
Let's not use `utils` as package name, this is a bad practice.
This package deals with zbus signal management, let's find a name that relates to that.

Reply:
Actually it's not related to zbus at all. it's just a generic context cancellation mechanism on signal receiving. I use it to handle signals to gracefully shutdown services.

may be rename this to `signal` ? I couldn't come up with a better name.